src/Host/Client/Impl/Extensions/RStringExtensions.cs

```diff
@@ -17,7 +17,18 @@ public static class RStringExtensions {
                 return nullValue;
             }
 
-            return quote + s.Replace("\\", "\\\\").Replace("" + quote, "\\" + quote) + quote;
+            return quote + s.Replace("\\", "\\\\").Replace("" + quote, "\\" + quote).EscapeCharacterConstants() + quote;
+        }
+
+        private static string EscapeCharacterConstants(this string s) {
+            s = s.Replace("\n", "\\n")
```

Comment:
This looks wrong. Consider this string as input :
```
\\n
```
After the first replace (in the method above) it becomes:
```
\\\\n
```
Then this replace will make it into
```
\\\\\n
```
And now when R parses it, it'll see `\\` `\\` `\n` - and `\n` becomes an unintended newline, instead of literal `\` followed by literal `n`.

I'm actually not clear why this is needed at all. The original code replaces all instances of `\` with `\\`, turning them into literal backslashes instead of the beginning of an escape sequence. So if input is `\n`, it becomes `\\n`, and then R will unescape `\\` and turn it back to `\n` when parsing, as intended. So there shouldn't be a need for any special handling for `\n` and other such escapes.